src/main/java/org/apache/maven/plugins/javadoc/resolver/ResourceResolver.java

```diff
@@ -415,14 +412,15 @@ private Collection<Path> resolveAndUnpack( final List<Artifact> artifacts,
                 if ( propagateErrors )
                 {
                     throw new ArtifactResolutionException( "Failed to retrieve valid un-archiver component: "
-                        + a.getType(), a, e );
+                        + a.getExtension(), RepositoryUtils.toArtifact( a ), e );
                 }
             }
             catch ( final ArchiverException e )
             {
                 if ( propagateErrors )
                 {
-                    throw new ArtifactResolutionException( "Failed to unpack: " + a.getId(), a, e );
+                    throw new ArtifactResolutionException( "Failed to unpack: " + a.toString(),

```

Comment:
`toString()` will be automatically invoked.

Reply:
Fixed in https://github.com/apache/maven-javadoc-plugin/pull/164/commits/b88e2980334497fbafa0d26dd48fcf136f819dd0.